packages/eddsa-poseidon/src/utils.ts

```diff
@@ -72,14 +74,15 @@ export function checkMessage(message: BigNumberish): bigint {
 }
 
 /**
- * Computes the Blake512 hash of the input message.
- * Blake512 is a cryptographic hash function that produces a hash value of 512 bits,
- * commonly used for data integrity checks and other cryptographic applications.
+ * Computes a hash of an input, given a specified hashing algorithm.
  * @param message The input data to hash, provided as a Buffer.
- * @returns A Buffer containing the 512-bit hash result.
+ * @param algorithm The selected algorithm
+ * @returns A Buffer containing the result
  */
-export function hash(message: Buffer | Uint8Array): Buffer {
-    const engine = new Blake512()
+export function hashInput(message: Buffer | Uint8Array, algorithm?: SupportedHashingAlgorithms) {
+    let engine: HashFunction
+    if (!algorithm || algorithm === SupportedHashingAlgorithms.BLAKE1) engine = new Blake512()
```

Comment:
If I pass an unsupported algorithm, this function would still return Blake2b right? I think it's better to use a `switch` here, where the default case throws an exception warning devs that the algorithm is not supported.

Reply:
good point, done!